### PR TITLE
OGM-1346 Add GeometryCollection to spatial support

### PR DESCRIPTION
**Pull request by Krerg**
* https://hibernate.atlassian.net/browse/OGM-1346

### CODE REVIEW

**Comment by fax4ever** [comment]
add to whitelist

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/AbstractGeoJsonObject.java [hunk 1]
```diff
@@ -20,21 +20,34 @@
...
  * @author Guillaume Smet
+ * @author Aleksandr Mylnikov
  */
 public abstract class AbstractGeoJsonObject implements Serializable {
 
 	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
 
+	/**
+	 * Name of key for object stores geo data.
+	 * For all geoobjects it's coordinates, but for {@link GeoCollection} is geometries
```
Comment (gsmet) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/AbstractGeoJsonObject.java]: Maybe:
```
Name of the main key used in the JSON document.
For most geo objects, it is {@code coordinates}. For {@link GeoCollection}, it is {@code geometries}.
```

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/GeoCollection.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	/**
+	 * Instantiates a new GeometryCollection.
+	 *
+	 * @param firstGeoObject       the first point
+	 * @param additionalGeoObjects the additional points
+	 */
+	public GeoCollection(AbstractGeoJsonObject firstGeoObject, AbstractGeoJsonObject... additionalGeoObjects) {
+		super( TYPE );
+		Contracts.assertNotNull( firstGeoObject, "firstPoint" );
+		Contracts.assertNotNull( additionalGeoObjects, "additionalPoints" );
```
Comment (gsmet) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/GeoCollection.java]: The parameter names should be changed in the 2 calls above.

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/AbstractGeoJsonObject.java (hunk 1 above)
Comment (Krerg) [reply mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/AbstractGeoJsonObject.java]: It's done.

**Comment by Krerg** [comment]
@gsmet It's done

**Comment by fax4ever** [comment]
Really nice work. Nothing to say about it.

**Comment by fax4ever** [comment]
Rebased and merged thank you #1042.

**Pull request closed without merging**